### DISCUSSION TITLE
NewConn vs NewPool

### DISCUSSION THREAD

**Discussion by nikitacrit** [pashagolub/pgxmock#83] [Q&A] [answered]
Hello! What is actually difference between NewConn vs NewPool? Any recommendation what should we use? Thanks!

**Comment by pashagolub** [comment, marked as answer]
Hello,

depends on your application code. If you are using [pgxpool](https://pkg.go.dev/github.com/jackc/pgx/v4/pgxpool) package then you probably want [func NewPool(options ...func(*pgxmock) error) (PgxPoolIface, error)](https://pkg.go.dev/github.com/pashagolub/pgxmock#NewPool).

If you are using [pgx](https://pkg.go.dev/github.com/jackc/pgx/v4) then you should use [func NewConn(options ...func(*pgxmock) error) (PgxConnIface, error)](https://pkg.go.dev/github.com/pashagolub/pgxmock#NewConn).

`*pgx.Conn` represents a single connection to the database and is not concurrency safe. Use sub-package `pgxpool` for a concurrency-safe connection pool.

Regards
